src/vignette/util/thumbnail.clj

```diff
@@ -79,6 +79,7 @@
 
 (defn get-or-generate-thumbnail
   [system thumb-map]
+  (assert-original-mime-type (get thumb-map :original) thumb-map)
```

Comment:
This just gets the mime type from a string? If so, it's relying on the file extension, which isn't always there. If these images pass this check, then I'd say LGTM